File: src/main/java/com/randomperson22/piggymodtest/entities/zompiggy/ZompiggyModel.java
package com.randomperson22.piggymodtest.entities.zompiggy;

import net.minecraft.util.ResourceLocation;
import software.bernie.geckolib3.core.event.predicate.AnimationEvent;
import software.bernie.geckolib3.core.processor.IBone;
import software.bernie.geckolib3.model.AnimatedGeoModel;

public class ZompiggyModel extends AnimatedGeoModel<EntityZompiggy> {
	
    public ResourceLocation getModelLocation(EntityZompiggy entity) {
        return new ResourceLocation("piggymodtest", "geo/zompiggy.geo.json");
    }

    public ResourceLocation getTextureLocation(EntityZompiggy entity) {
        return new ResourceLocation("piggymodtest", "textures/entities/zompiggy.png");
    }

    public ResourceLocation getAnimationFileLocation(EntityZompiggy entity) {
        return new ResourceLocation("piggymodtest", "animations/zompiggy.animation.json");
    }
}

File: src/main/kotlin/com/randomperson22/piggymodtest/entities/zompiggy/ZompiggyModel.kt
package com.randomperson22.piggymodtest.entities.zompiggy

import net.minecraft.util.ResourceLocation
import software.bernie.geckolib3.model.AnimatedGeoModel

class ZompiggyModel : AnimatedGeoModel<EntityZompiggy?>() {
    override fun getModelLocation(entity: EntityZompiggy?): ResourceLocation {
        return ResourceLocation("piggymodtest", "geo/zompiggy.geo.json")
    }

    override fun getTextureLocation(entity: EntityZompiggy?): ResourceLocation {
        return ResourceLocation("piggymodtest", "textures/entities/zompiggy.png")
    }

    override fun getAnimationFileLocation(entity: EntityZompiggy?): ResourceLocation {
        return ResourceLocation("piggymodtest", "animations/zompiggy.animation.json")
    }
}
